### PR TITLE
Enable  qwenimage/qwenimage_edit/qwenimage_edit_plus on hpu

### PR DESCRIPTION
**Pull request by yingjie-han**
Test example for "Qwen/Qwen-Image-Edit-2509":

```
import os
os.environ['PT_HPU_LAZY_MODE'] = "1"
os.environ['QWEN25VL_FP32_SOFTMAX'] = "True"

import torch
import time
from PIL import Image

from optimum.habana.diffusers import GaudiQwenImageEditPlusPipeline
from optimum.habana.transformers.modeling_utils import adapt_transformers_to_gaudi
from optimum.habana.transformers.gaudi_configuration import GaudiConfig

adapt_transformers_to_gaudi()

gaudi_config_kwargs = {"use_fused_adam": True, "use_fused_clip_norm": True}
gaudi_config_kwargs["use_torch_autocast"] = True
gaudi_config = GaudiConfig(**gaudi_config_kwargs)

pipeline = GaudiQwenImageEditPlusPipeline.from_pretrained(       
        "Qwen/Qwen-Image-Edit-2509",
        torch_dtype=torch.bfloat16,
        use_habana=True,
        use_hpu_graphs=False,
        gaudi_config=gaudi_config,
        )
pipeline.set_progress_bar_config(disable=None)

image1 = Image.open("duck.jpg")
image2 = Image.open("yarn-art-pikachu.png")
prompt = "The duck is on the right, the pikachu is on the left, and they are fighting."
inputs = {
        "image": [image1, image2],
        "prompt": prompt,
        "generator": torch.manual_seed(0),
        "true_cfg_scale": 4.0,
        "negative_prompt": " ",
        "num_inference_steps": 40,
        "guidance_scale": 1.0,
}

with torch.inference_mode():
        output = pipeline(**inputs)
        output_image = output.images[0]
        output_image.save("output_image_edit_2509.png")
```

### CODE REVIEW

File: optimum/habana/diffusers/models/attention_processor.py [hunk 1]
```diff
@@ -534,5 +534,119 @@ def __call__(
...
+        self,
+        attn: Attention,
+        hidden_states: torch.FloatTensor,  # Image stream
+        encoder_hidden_states: torch.FloatTensor = None,  # Text stream
+        encoder_hidden_states_mask: torch.FloatTensor = None,
+        attention_mask: Optional[torch.FloatTensor] = None,
+        image_rotary_emb: Optional[torch.Tensor] = None,
+    ) -> torch.FloatTensor:
+        if encoder_hidden_states is None:
+            raise image_rotary_embValueError("QwenDoubleStreamAttnProcessor2_0 requires encoder_hidden_states (text stream)")
```
Comment (miaojinc) [inline optimum/habana/diffusers/models/attention_processor.py]: Is this typo?

File: optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_plus.py [hunk 2]
```diff
@@ -0,0 +1,700 @@
...
+        config = self.transformer.config
+        self.transformer.pos_embed = GaudiQwenEmbedRope(theta=10000, axes_dim=list(config['axes_dims_rope']), scale_rope=True)              
+            
+        if use_hpu_graphs:
+            from habana_frameworks.torch.hpu import wrap_in_hpu_graph
+            #self.transformer = wrap_in_hpu_graph(self.transformer)
+            for block in self.transformer.transformer_blocks:
+                block = wrap_in_hpu_graph(block)
+            
+            # self.text_encoder = wrap_in_hpu_graph(self.text_encoder)
```
Comment (Wei-Lin-Intel) [inline optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_plus.py]: Either removing this comment, or use a parameter to control using hpu graph.

File: optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_plus.py [hunk 3]
```diff
@@ -0,0 +1,700 @@
...
+            return_tensors="pt",
+        ).to(device)
+
+        outputs = self.text_encoder(
+            input_ids=model_inputs.input_ids,
+            attention_mask=model_inputs.attention_mask,
+            pixel_values=model_inputs.pixel_values,
+            image_grid_thw=model_inputs.image_grid_thw,
+            output_hidden_states=True,
+            use_flash_attention=True,
```
Comment (Wei-Lin-Intel) [inline optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_plus.py:315]: It is better to apply fp32 SDPA to OH Qwen2_5_vl to gurantee the accuracy.

**Comment by Wei-Lin-Intel** [comment]
Please apply `make style` to reformat

File: optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage.py [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+        >>> # Depending on the variant being used, the pipeline call will slightly vary.
+        >>> # Refer to the pipeline documentation for more details.
+        >>> image = pipe(prompt, num_inference_steps=50).images[0]
+        >>> image.save("qwenimage.png")
+        ```
+"""
+
+class GaudiQwenImagePipeline(GaudiDiffusionPipeline, QwenImagePipeline):
+    r"""
+    Adapted from: https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L132
```
Comment (mengker33) [inline optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage.py]: It is better to use a specific commit or tag for the code link you refer to, as "main" always points to the latest and gets updated frequently. for example use link :https://github.com/huggingface/diffusers/blob/commit_id_xxx/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L132

File: optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit.py [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
+from diffusers.schedulers import FlowMatchEulerDiscreteScheduler
+from diffusers.utils import logging, replace_example_docstring
+from diffusers.pipelines.qwenimage.pipeline_output import QwenImagePipelineOutput
+from ....transformers.gaudi_configuration import GaudiConfig
+from ..pipeline_utils import GaudiDiffusionPipeline
+from diffusers.pipelines.qwenimage.pipeline_qwenimage_edit import QwenImageEditPipeline,calculate_dimensions,calculate_shift,retrieve_timesteps
+from ...models.attention_processor import GaudiQwenDoubleStreamAttnProcessor2_0
+from ...models.qwenimage_transformer import QwenImageTransformer2DModelGaudi
+from ....utils import HabanaProfile
+from ....transformers.modeling_utils import adapt_transformers_to_gaudi
```
Comment (mengker33) [inline optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit.py]: Did you use adapt_transformers_to_gaudi, GaudiQwen2_5_VLForConditionalGeneration and GaudiQwen2_5_VLModel anywhere is qwenimage_edit? If not, please remove the irrelevant imports :)

File: optimum/habana/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py [hunk 6]
```diff
@@ -834,7 +834,7 @@ def forward(
         rope_deltas: Optional[torch.LongTensor] = None,
         cache_position: Optional[torch.LongTensor] = None,
         second_per_grid_ts: Optional[torch.Tensor] = None,
-        use_flash_attention: Optional[bool] = False,
+        use_flash_attention: Optional[bool] = True,
```
Comment (mengker33) [inline optimum/habana/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py]: Looks like these changes not related to qwen image enabling?

Comment (yingjie-han) [reply optimum/habana/transformers/models/qwen2_5_vl/modeling_qwen2_5_vl.py]: This is the old code, outdated, in latest code qwen2_5vl is not changed. I set the input param use_flash_attention=True for in qwenimage's text_encoder , instead of change the default input value of qwen2_5vl.

File: examples/stable-diffusion/text_to_image_qwenimage.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+from optimum.habana.transformers.gaudi_configuration import GaudiConfig
+
+from optimum.habana.diffusers import GaudiQwenImagePipeline
+import time
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--model_name_or_path",
+        #default="Qwen/Qwen-Image",
```
Comment (mengker33) [inline examples/stable-diffusion/text_to_image_qwenimage.py]: Should default one be "Qwen/Qwen-Image"?

Comment (yingjie-han) [reply examples/stable-diffusion/text_to_image_qwenimage.py]: Yes, it is typo.

**Comment by yingjie-han** [comment]
> Please apply `make style` to reformat
Done.

File: optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage.py (hunk 4 above)
Comment (yingjie-han) [reply optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage.py]: fixed. 3ks.

File: optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit.py (hunk 5 above)
Comment (yingjie-han) [reply optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit.py]: fixed.3ks

File: optimum/habana/diffusers/models/attention_processor.py (hunk 1 above)
Comment (yingjie-han) [reply optimum/habana/diffusers/models/attention_processor.py]: Yes，fixed. Thanks!

File: optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_plus.py (hunk 3 above)
Comment (Wei-Lin-Intel) [reply optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_plus.py:315]: Better to set the flag as the default.

File: optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_plus.py (hunk 2 above)
Comment (yingjie-han) [reply optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_plus.py]: fixed.

File: optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_plus.py (hunk 3 above)
Comment (yingjie-han) [reply optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit_plus.py:315]: os.environ["QWEN25VL_FP32_SOFTMAX"] = "True" is set in the __init__ function of pipeline.

**Comment by yingjie-han** [comment]
@Wei-Lin-Intel @mengker33 @huijuanzh please help to review the multicards code for qwenimage. The perf. test on G2E is:
image
image

**Comment by huijuanzh** [comment]
> @Wei-Lin-Intel @mengker33 @huijuanzh please help to review the multicards code for qwenimage. The perf. test on G2E is: image image

Do we have test with --use_hpu_graphs ?

**Comment by yingjie-han** [comment]
> Do we have test with --use_hpu_graphs ?

@huijuanzh --use_hpu_graphs  will have OOM problem. And it only have about 5% perf. gain. So we don't use hpu_graphs .

**Comment by huijuanzh** [comment]
The performance data seems multi-card scaling is not good, others LGTM

**Comment by Wei-Lin-Intel** [comment]
Please update diffusers to the latest release, and resolve the code conflict.

**Comment by yingjie-han** [comment]
> Please update diffusers to the latest release, and resolve the code conflict.

@Wei-Lin-Intel Done
